tests/launch/test_launch.py

```diff
@@ -528,12 +528,25 @@ def test_launch_agent(
     assert len(ctx["launch_agents"].keys()) == 1
 
 
+def test_launch_agent_instance(test_settings, live_mock_server):
+    api = wandb.sdk.internal.internal_api.Api(
+        default_settings=test_settings, load_settings=False
+    )
+    agent = LaunchAgent("mock_server_entity", "test_project", ["default"])
+    ctx = live_mock_server.get_ctx()
+    assert len(ctx["launch_agents"]) == 1
+    assert agent._id == int(list(ctx["launch_agents"].keys())[0])
+
+    get_agent_response = api.get_launch_agent(agent._id, agent.gorilla_supports_agents)
+    assert get_agent_response["name"] == "test_agent"
+
+
 def test_launch_agent_different_project_in_spec(
     test_settings,
     live_mock_server,
     mocked_fetchable_git_repo,
     monkeypatch,
-    mock_load_backend_agent,
+    # mock_load_backend_agent,
```

Comment:
took this out because it was breaking agent functionality (wasn't previously because we used to rely on a singlethreaded logic flow in the agent)

Reply:
That's fine. This can just be deleted.